ext/dep_gecode/dep_selector_to_gecode.cpp

```diff
@@ -843,7 +843,9 @@ int VersionProblem::Solve(VersionProblem * problem, VersionProblem ** solution)
         DEBUG_STREAM << problem->DebugPrefix() << "Solver Best Solution " << *solution << std::endl << std::flush;
     }
 
-    pool->Delete(*solution);
+    if (*solution) {
+        pool->Delete(*solution);
+    }
```

Comment:
@markan , line#843 just above: `DEBUG_STREAM << problem->DebugPrefix() << "Solver Best Solution " << *solution << std::endl << std::flush;`, outputs `*solution` as 0 in case no solution is found. Following lines also output the final status: https://github.com/chef/dep-selector/blob/master/ext/dep_gecode/dep_selector_to_gecode.cpp#L814-L817. This logging happens when `Debugmodeon` file is created.  

If `Debugmodeon` file is not created, user gets either `The dependency constraints could not be solved in the time allotted.` or `Unable to satisfy constraints on package <pkg_name> due to solution constraint <constraint>.`